### PR TITLE
Docker Image: Fix release tag in publish workflow

### LINKED ISSUE #6510: AIIDA_VERSION was removed from build.json
https://github.com/aiidateam/aiida-core/blob/e91371573a84d4a68d6107f33c392b8718f2f26f/.github/workflows/docker-publish.yml#L72

It cause the problem that aiida version tag not exist in https://github.com/aiidateam/aiida-core/pkgs/container/aiida-core-with-services @danielhollas

### PR DESCRIPTION
**Pull request by danielhollas**
Fixes #6510.

Looks like this works, see this [test run](https://github.com/aiidateam/aiida-core/actions/runs/9807907715?pr=6520) which generated the [aiida-2.6.x-test](https://github.com/aiidateam/aiida-core/pkgs/container/aiida-core-base/239748429?tag=aiida-2.6.x-test) tag on ghcr.io.

### CODE REVIEW

**Comment by sphuber** [comment]
Looks good to me. @unkcpz can you also have a quick check to see that this makes sense? Thanks

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -59,6 +59,12 @@ jobs:
         vars=$(cat build.json | jq -c '[.variable | to_entries[] | {"key": .key, "value": .value.default}] | from_entries')
         echo "vars=$vars" | tee -a "${GITHUB_OUTPUT}"
 
+    - id: get-version
+      if: ${{ github.ref_type == 'tag'  && startsWith(github.ref_name, 'v') }}
+      run: |
+        tag="${{ github.ref_name }}"
+        echo "AIIDA_VERSION=${tag#v}" >> $GITHUB_OUTPUT
```
Comment (unkcpz) [inline .github/workflows/docker-publish.yml:66]: I assume `${tag#v}` is drop `v` in front, correct?

**Review by unkcpz — APPROVED** [review]
Looks good to me.

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (danielhollas) [reply .github/workflows/docker-publish.yml:66]: Yep.

**Pull request merged**